### PR TITLE
Throw more specific error, ArgumentError, rather than a general Error()

### PR DESCRIPTION
**Pull request by petershintech**
Also changed the unit test to check ArgumentError, not ErrorException.
#1867

### CODE REVIEW

File: src/dataframe/dataframe.jl [hunk 1]
```diff
@@ -786,7 +786,7 @@ function insertcols!(df::DataFrame, col_ind::Int, name_col::Pair{Symbol, <:Abstr
                      makeunique::Bool=false)
     name, item = name_col
     0 < col_ind <= ncol(df) + 1 || throw(BoundsError())
-    size(df, 1) == length(item) || size(df, 2) == 0 || error("number of rows does not match")
+    size(df, 1) == length(item) || size(df, 2) == 0 || throw(ArgumentError("number of rows does not match"))
```
Comment (bkamins) [inline src/dataframe/dataframe.jl]:
```suggestion
    if !(size(df, 1) == length(item) || size(df, 2) == 0)
        throw(ArgumentError("number of rows does not match"))
    end
```

Comment (bkamins) [reply src/dataframe/dataframe.jl]: this fixes the overlong line

**Comment by petershintech** [comment]
@bkamins Thanks. I agree your code is easy to read. In fact, I can see several long lines exploiting short circuits. Do you think we need to rewrite the lines to more readable ones? If so, I can help you.

File: src/dataframe/dataframe.jl [hunk 2]
```diff
@@ -786,7 +786,9 @@ function insertcols!(df::DataFrame, col_ind::Int, name_col::Pair{Symbol, <:Abstr
                      makeunique::Bool=false)
     name, item = name_col
     0 < col_ind <= ncol(df) + 1 || throw(BoundsError())
-    size(df, 1) == length(item) || size(df, 2) == 0 || error("number of rows does not match")
+    if !(size(df, 1) == length(item) || size(df, 2) == 0)
+        throw(ArgumentError("number of rows does not match"))
```
Comment (nalimilan) [inline src/dataframe/dataframe.jl]:
```suggestion
        throw(ArgumentError("length of new column ($(length(item))) must match the number of rows in data frame ($(nrow(df)))"))
```

**Comment by bkamins** [comment]
>  I can see several long lines exploiting short circuits.

Sure - thank you. The "soft" rule is that lines should not be longer than 92 characters.

**Comment by petershintech** [comment]
I like the new message about more specific about the cause. Commit to the branch and ready to be merged. Thanks.

**Comment by petershintech** [comment]
One quick question before merging. In this case,
if !(size(df, 1) == length(item) || size(df, 2) == 0)
should I throw ArgumentError or DimensionMismatch? I think the latter one is more specific about error cause so it would be better than the first one. What do you think of it?

**Comment by nalimilan** [comment]
Right, `DimensionMismatch` sounds better.

**Comment by petershintech** [comment]
Ok. I have replaced ArgumentError with DimensionMismatch and also touched a unit test for it. I can see the same change would be desirable for some other code lines but it can be captured in a separate PR. I think it is ready to merge. Thanks for all your inputs.

**Comment by bkamins** [comment]
Yes, we used `ArgumentError` or `BoundsError` only everywhere up till now. We can be more specific of course 😄. Thank you.

**Pull request merged**